[DOCS-5315] Add Suppressions doc (#21145)
DataDog/documentation: content/en/security/suppressions.md

---
title: Suppressions
kind: documentation
disable_toc: false
further_reading:
- link: "security/detection_rules/"
  tag: "Documentation"
  text: "Learn more about detection rules"
---

<div class="alert alert-warning">Suppressions are available for Cloud Workload Security and Cloud SIEM.</div>

## Overview

Suppressions are specific conditions for when a signal should not be generated, which can improve the accuracy and relevance of the signals that are generated.

## Suppression routes

> Suppressions can be set up for individual detection rules and also more broadly across multiple detection rules with suppression rules.

Author: Thanks @jhgilbert! The new suppression rule, which also has a suppression query, is available in the detection rule, but the legacy standalone suppression query is what is going to be deprecated.

@nimishaaa I made a couple of changes to try and clarify the legacy suppression query:

- Added an image to the migration section
- Also added a yellow banner at the top of the doc about the deprecation with a link to the migration section.

Could you let me know if this works?

### Detection rules

When you [create][1] or [modify][2] a detection rule, you can define a suppression query to prevent a signal from getting generated. For example, add a rule query to determine when a detection rule triggers a security signal. You can also customize the suppression query to suppress signals for a specific attribute value.

{{< img src="security/security_monitoring/suppressions/detection_suppression_rule.png" alt="The detection rule editor showing the add suppression query section" style="width:65%;" >}}

### Suppression rules

> Use suppression rules to set general suppression conditions across multiple detection rules and signals instead of setting up suppression conditions for each individual detection rules. For example, you can set up a suppression rule to suppress any signal that contains a specific IP.

Reviewer: small nit: Use suppression rules to set general suppression conditions across multiple detection rules ~~and signals~~ instead of setting up suppression conditions for each individual detection rules.

Author: Great, thanks, I was wondering about that!

> Use suppression rules to set general suppression conditions across multiple detection rules and signals instead of setting up suppression conditions for each individual detection rules. For example, you can set up a suppression rule to suppress any signal that contains a specific IP.

Author: ```suggestion
Use suppression rules to set general suppression conditions across multiple detection rules instead of setting up suppression conditions for each individual detection rules. For example, you can set up a suppression rule to suppress any signal that contains a specific IP.
```

## Supressions configuration

### Suppression list

The [suppression list][3] provides a centralized and organized way for you to manage suppressions across multiple detection rules.

{{< img src="security/security_monitoring/suppressions/suppression_list.png" alt="The suppressions page showing a list of suppression rules" style="width:90%;" >}}

## Create a suppression

> 1. Navigate to the [Suppressions][3] page.

Reviewer 2: I'm getting a 404 for the Suppressions page, but that may be expected for now.

Author: Yes expected, the feature isn't out yet.

1. Click **+ New Suppression**.
1. Enter a name for the suppression query.
1. Add a description to provide context on why this suppression is being applied.
1. Optionally, add an expiration date on which this suppression will be deactivated.
1. Select the detection rules you want to apply this suppression to. You can select multiple detection rules.
1. In the **Add Suppression Query** section, you have the option to enter suppression queries so that a signal is not generated when the values are met. For example, if a user `john.doe` is triggering a signal, but their actions are benign and you no longer want signals triggered from this user, input the log query: `@user.username:john.doe`.

{{< img src="security/security_monitoring/suppressions/suppression_query.png" alt="The add suppression query with the query @user.username:john.doe" style="width:65%;" >}}

## Migrate suppression queries to suppression rules

> <div class="alert alert-warning">Suppression rules are replacing suppression queries. Suppression queries will be deprecated on April 1, 2024.</div>

Author: @nimishaaa does this work? Suppression queries are used in the suppression rules so I wanted to clarify that the legacy suppression query is related to detection rules.

```suggestion
<div class="alert alert-warning">Suppression rules are replacing suppression queries in detection rules. The legacy suppression queries will be deprecated on April 1, 2024.</div>
```

Reviewer: yes LGTM - i can see how the naming can be confusing. thanks for clearing it up! 🙇🏽‍♀️

Migrate your legacy suppression queries to suppression rules. To see a list of rules using the legacy suppression and to migrate them:
1. Navigate to the [detection rules list][4].

> 1. Hover over the **xx rules** in the yellow banner to see the list of rules that need to be migrated.

Reviewer 2: ```suggestion
1. Hover over **xx rules** in the yellow banner to see the list of rules that need to be migrated.
```

---
title: Suppressions
kind: documentation
disable_toc: false
further_reading:
- link: "security/detection_rules/"
  tag: "Documentation"
  text: "Learn more about detection rules"
---

<div class="alert alert-warning">Suppressions are available for Cloud Security Management Threats and Cloud SIEM.</div>

<div class="alert alert-warning">Suppression rules are replacing suppression queries in detection rules. The legacy suppression queries will be deprecated on April 1, 2024. See <a href="#migrate-legacy-suppression-queries-to-suppression-rules">Migrate legacy suppression queries to suppression rules</a> for more information. </div>

## Overview

Suppressions are specific conditions for when a signal should not be generated, which can improve the accuracy and relevance of the signals that are generated.

## Suppression routes

> You can set up suppressions within individual [detection rules](#detection-rules), or define a separate [suppression rule](#suppression-rules) to suppress signals more broadly across multiple detection rules.

Reviewer 2: ```suggestion
You can set up a suppression query within an individual [detection rule](#detection-rules), or define a separate [suppression rule](#suppression-rules) to suppress signals across one or more detection rules.
```

### Detection rules

> 

Reviewer 2: You might move the warning to here, since the user won't have much context for understanding the warning if it's at the top.

Author: Good idea! done.
